File: api-python/routers/seed.py
from fastapi import APIRouter
from pydantic import BaseModel

router = APIRouter(tags=["seed"])

seed_number = 42


class SeedUpdate(BaseModel):
    seedNumber: int

    class Config:
        json_schema_extra = {"example": {"seedNumber": 100}}


@router.get("/seed")
async def get_seed() -> dict[str, int]:
    return {"seedNumber": seed_number}


@router.post("/update-seed")
async def update_seed(seed_data: SeedUpdate) -> dict[str, int]:
    global seed_number
    seed_number = seed_data.seedNumber
    return {"seedNumber": seed_number}

File: api/api-go/internal/graphql_server/utils/seed.go
package utils

type Seed struct {
	SeedNumber int `json:"seedNumber"`
}

var seedNumber = 42

func GetSeed() (*Seed, error) {
	return &Seed{
		SeedNumber: seedNumber,
	}, nil
}

func SetSeed(n int) (*Seed, error) {
	seedNumber = n
	return &Seed{
		SeedNumber: seedNumber,
	}, nil
}
